ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/MLSdkAsyncHttpResponseHandler.java

```diff
@@ -0,0 +1,221 @@
+/*
+ *
+ *  * Copyright OpenSearch Contributors
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.opensearch.ml.engine.algorithms.remote;
+
+import static org.opensearch.ml.engine.algorithms.remote.ConnectorUtils.processErrorResponse;
+import static org.opensearch.ml.engine.algorithms.remote.ConnectorUtils.processOutput;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.apache.http.HttpStatus;
+import org.apache.logging.log4j.util.Strings;
+import org.opensearch.OpenSearchStatusException;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.ml.common.connector.Connector;
+import org.opensearch.ml.common.output.model.ModelTensors;
+import org.opensearch.script.ScriptService;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import lombok.Getter;
+import lombok.extern.log4j.Log4j2;
+import software.amazon.awssdk.http.SdkHttpFullResponse;
+import software.amazon.awssdk.http.SdkHttpResponse;
+import software.amazon.awssdk.http.async.SdkAsyncHttpResponseHandler;
+
+@Log4j2
+public class MLSdkAsyncHttpResponseHandler implements SdkAsyncHttpResponseHandler {
+    @Getter
+    private Integer statusCode;
+    @Getter
+    private final StringBuilder responseBody = new StringBuilder();
+
+    private WrappedCountDownLatch countDownLatch;
+
+    private ActionListener<List<ModelTensors>> actionListener;
+
+    private Map<String, String> parameters;
+
+    private Map<Integer, ModelTensors> tensorOutputs;
+
+    private Connector connector;
+
+    private ScriptService scriptService;
+
+    private final static Gson GSON = new GsonBuilder().disableHtmlEscaping().create();
```

Comment:
Can't we re-use this [class](https://github.com/opensearch-project/ml-commons/blob/main/common/src/main/java/org/opensearch/ml/common/utils/StringUtils.java#L43)?

Reply:
No, we have to make the gson `disableHtmlEscaping` to ensure not breaking our ITs.

Reply:
I don't remember we have `disableHtmlEscaping` before , why it starts failing now?